nexus/src/app/sagas/instance_create.rs

```diff
@@ -237,106 +339,108 @@ async fn sic_alloc_server(
 /// Create a network interface for an instance, using the parameters at index
 /// `nic_index`, returning the UUID for the NIC (or None).
 async fn sic_create_network_interface(
-    sagactx: ActionContext<SagaInstanceCreate>,
-    nic_index: usize,
+    sagactx: NexusActionContext,
 ) -> Result<(), ActionError> {
-    let saga_params = sagactx.saga_params();
+    let repeat_saga_params = sagactx.saga_params::<NetParams>()?;
+    let saga_params = repeat_saga_params.saga_params;
+    let nic_index = repeat_saga_params.which;
+    let instance_id = repeat_saga_params.instance_id;
+    let interface_id = repeat_saga_params.new_id;
     let interface_params = &saga_params.create_params.network_interfaces;
     match interface_params {
         params::InstanceNetworkInterfaceAttachment::None => Ok(()),
         params::InstanceNetworkInterfaceAttachment::Default => {
-            sic_create_default_primary_network_interface(&sagactx, nic_index)
-                .await
+            sic_create_default_primary_network_interface(
+                &sagactx,
+                &saga_params,
+                nic_index,
+                instance_id,
+                interface_id,
+            )
+            .await
         }
         params::InstanceNetworkInterfaceAttachment::Create(
             ref create_params,
         ) => match create_params.get(nic_index) {
             None => Ok(()),
             Some(ref prs) => {
-                sic_create_custom_network_interface(&sagactx, nic_index, prs)
-                    .await
+                sic_create_custom_network_interface(
+                    &sagactx,
+                    &saga_params,
+                    instance_id,
+                    interface_id,
+                    prs,
+                )
+                .await
             }
         },
     }
 }
 
 /// Delete one network interface, by index.
```

Comment:
The comment here, deleting by index, is no longer accurate, I think. It's done by primary key / ID.

Reply:
Nice catch.  Fixed in 96dba6366f6cb8b47ff10a3a4b0bf0277f9d060e.